tests/unit/schema/test_schema_logic.py

```diff
@@ -0,0 +1,202 @@
+    format_param,
+    generate_schema,
+    get_params_from_command,
+    retrieve_command_structure,
+)
+
+
+class TestParameterSchema(TestCase):
+    @parameterized.expand(
+        [
+            ("", "", {}),
+            ("default", "default value", {"default": "default value"}),
+            ("items", "item type", {"items": {"type": "item type"}}),
+            ("choices", ["1", "2"], {"enum": ["1", "2"]}),
+        ]
+    )
+    def test_parameter_to_schema(self, property_name, property_value, added_property_field):
+        param = SamCliParameterSchema("param name", "param type", "param description")
+        param.__setattr__(property_name, property_value)
+
+        param_schema = param.to_schema()
+
+        expected_schema = {"title": "param name", "type": "param type", "description": "param description"}
+        expected_schema.update(added_property_field)
+        self.assertEqual(expected_schema, param_schema)
+
+
+class TestCommandSchema(TestCase):
+    def test_command_to_schema(self):
+        params = [SamCliParameterSchema("param1", "string"), SamCliParameterSchema("param2", "number")]
+        command = SamCliCommandSchema("commandname", "command description", params)
+        command_schema = command.to_schema()
+        self.assertEqual(len(command_schema.keys()), 1)
+        self.assertEqual(list(command_schema.keys())[0], "commandname")
+        inner_schema = command_schema["commandname"]
+        for expected_key in ["title", "description", "properties", "required"]:
+            self.assertIn(expected_key, inner_schema.keys(), f"Command schema should have key {expected_key}")
+        self.assertIn("parameters", inner_schema["properties"].keys(), "Schema should have 'parameters'")
+        for expected_key in ["title", "description", "type", "properties"]:
+            self.assertIn(
+                expected_key,
+                inner_schema["properties"]["parameters"],
+                f"Parameters schema should have key {expected_key}",
+            )
+        for param in params:
+            self.assertIn(
+                param.name, inner_schema["properties"]["parameters"]["properties"], f"{param.name} should be in schema"
+            )
+            self.assertEqual(
+                param.to_schema(),
+                inner_schema["properties"]["parameters"]["properties"].get(param.name),
+                f"{param.name} should point to schema representation",
+            )
+        self.assertEqual(["parameters"], inner_schema["required"], "Parameters attribute should be required")
+
+
+class TestSchemaLogic(TestCase):
+    @parameterized.expand(
+        [
+            ("string", "string"),
+            ("integer", "integer"),
+            ("number", "number"),
+            ("text", "string"),
+            ("path", "string"),
+            ("choice", "string"),
+            ("filename", "string"),
+            ("directory", "string"),
+            ("LIST", "array"),
+            (None, "string"),
```

Comment:
Ideally there shouldn't be any `None` parameters, I put it in mostly just as a safeguard in case something arises where it gets passed in. My logic is that, if something happens where a parameter is expected to have a `None` value, it might be best to label it as a string. But I can also see the value in altering this so that an exception of some sort gets thrown in case it does make its way in there, so it doesn't fail silently. I'd be open to either approach

Reply:
Lets raise an exception (or something that would stop the script gracefully) for this case. Mainly just want to remain on the side of caution and not include something that we don't want accidentally in the schema that we don't intend on adding. This way if there are cases of `None`, we can handle them manually.